### ISSUE TITLE
problems when merging BAM files of archaic samples

### ISSUE THREAD

**Issue by Overcraft90** [samtools/samtools#2037]
#### Are you using the latest version of [samtools](https://github.com/samtools/samtools/releases/latest) and [HTSlib](https://github.com/samtools/htslib/releases/latest)? If not, please specify.
samtools: `samtools 1.18-12-g147ac2f`
HTSlib: `htslib 1.18-50-g99415e2a`

 #### Please describe your environment.
- OS: `Linux 4.18.0-425.19.2.el8_7.x86_64`
- machine architecture: `x86_64`
- compiler:
```
gcc (GCC) 8.5.0 20210514 (Red Hat 8.5.0-16)
Copyright (C) 2018 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```
#### Please specify the steps taken to generate the issue, the command you are running and the relevant output.
After downloading five BAM files for Neanderthal, I proceeded to sort them:
```
samtools sort -n nea_<#>.bam -@ 112 -o nea_<#>_sorted.bam
```
and re-headed them with the following:
```
samtools view -H nea_<#>_sorted.bam | grep -v "^@RG" | samtools reheader - nea_<#>_sorted.bam > nea_<#>_sorted_reheaded.bam
```
before moving on and merge them as follow:
```
samtools merge -n -f -l 9 -o neanderthal.bam -@ 112 nea_1_sorted_reheaded.bam nea_2_sorted_reheaded.bam nea_3_sorted_reheaded.bam nea_4_sorted_reheaded.bam nea_5_sorted_reheaded.bam
```
However, upon launching the command for the `merge` I was prompted with the following

> [bam_translate] RG tag "L9302" on read "SN7001204_0130_AC0M6HACXX_PEdi_SS_L9302_L9303_1:1:1101:1050:3313" encountered with no corresponding entry in header, tag lost. Unknown tags are only reported once per input file for each tag ID.
> [bam_translate] RG tag "L9105" on read "NIOBE_0139_A_D0B5GACXX:6:1101:1227:3642" encountered with no corresponding entry in header, tag lost. Unknown tags are only reported once per input file for each tag ID.
> [bam_translate] RG tag "L9198" on read "SN928_0068_BB022WACXX:1:1101:1094:3181" encountered with no corresponding entry in header, tag lost. Unknown tags are only reported once per input file for each tag ID.
> [bam_translate] RG tag "L9199" on read "SN928_0073_BD0J78ACXX:1:1101:1249:195244" encountered with no corresponding entry in header, tag lost. Unknown tags are only reported once per input file for each tag ID.
> [bam_translate] RG tag "L9303" on read "SN7001204_0130_AC0M6HACXX_PEdi_SS_L9302_L9303_1:1:1101:1050:9131" encountered with no corresponding entry in header, tag lost. Unknown tags are only reported once per input file for each tag ID.

which I don't understand to what is related to... is it something I should be concerned with? The next step will be extracting the reads for this sample, so I'm considering any potential information loss. Let me know, thank in advance!

**Comment by jkbonfield** [comment]
It's saying that you have `RG:Z:` tags on your alignments but no corresponding `@RG` header line.  You could do a `samtools view -H file.bam > file.hdr`, manually edit it, and then use `samtools reheader` to fix your files.

Whether or not the issue matters depends on whether anything is reading the RG header to get extra information, such as library or machine type.  If so, you need to know this to edit the headers, and if you don't know it then quite possibly it's not helpful to try and silence these warnings anyway.

**Comment by Overcraft90** [comment]
@jkbonfield Indeed, I see. I did as you suggested but the problem persisted... unfortunately, I think those BAM files I downloaded from the ENA browser were somehow corrupted — albeit useful since the included all reads both mapped and unmapped.

In the end, I went back to the source at the EVA Max Planck website which is the one from where the files came from; it has been a bit of a pain since they split the BAM in 25 parts (auto + sex + mito) and download was very slow; however, the files were sorted and not corrupted, so I could move on and merge them.
Thanks again for the suggestion, I think I can close for now!
